cesium/internal/unary/delete.go

```diff
@@ -23,45 +26,55 @@ import (
 // In this case, only data after tr.Start from that pointer will be deleted, the
 // startOffset passed to domain will be calculated via db.index().Distance().
 //
+// Case 3 (edge case): |-----data-----|     *
+// This case only happens when the deletion start is after ALL known domains, therefore
+// we delete nothing.
+//
 // The same goes for the end pointer, but in the opposite direction (pointer will be
 // before or contains tr.End):
 //
 // Case 1 (* denotes tr.End):   |---------data---------|    *
 // In this case, that entire pointer will be deleted, and tr.End will be set to the
-// end of that pointer. The endOffset passed to domain will be the pointer's length.
+// end of that pointer. The endOffset passed to domain will 0.
 //
 // Case 2 (* denotes tr.End):   |----------data-----*----|
 // In this case, only data before tr.End from that pointer will be deleted, the
 // endOffset passed to domain will be calculated via db.index().Distance().
-func (db *DB) Delete(ctx context.Context, tr telem.TimeRange) error {
+//
+// Case 3 (edge case): |----data-----|   *
+// This case only happens when the deletion end is before (or equal) all known domains,
+// therefore we delete nothing.
+func (db *DB) delete(ctx context.Context, tr telem.TimeRange) error {
+	if !tr.Valid() {
+		return errors.Newf("delete start %d cannot be after delete end %d", tr.Start, tr.End)
+	}
+
 	var (
 		startOffset int64 = 0
 		endOffset   int64 = 0
+		density           = db.Channel.DataType.Density()
 	)
 
-	g, _, err := db.Controller.OpenAbsoluteGateIfUncontrolled(tr, control.Subject{Key: "Delete Writer"}, func() (controlledWriter, error) {
-		return controlledWriter{
-			Writer:     nil,
-			channelKey: db.Channel.Key,
-		}, nil
-	})
-
+	g, _, err := db.Controller.OpenAbsoluteGateIfUncontrolled(
+		tr,
+		control.Subject{Key: "delete_writer_" + uuid.New().String()},
```

Comment:
My take would be to leave the key as a UUID and then set the name to `delete_writer`.